### PR TITLE
Update valid and deprecated color-keywords to CSS color module level 3

### PR DESCRIPTION
**Pull request by bgriffith**
Currently the valid color keywords are only those specified in `CSS color module level 2.1`.

This is resulting in valid colors being incorrectly highlighted as deprecated.

[CSS color module level 3](http://www.w3.org/TR/css3-color/) has been a recommendation for a while now so I've updated the [list](http://www.w3.org/TR/css3-color/#svg-color) and updated the [deprecated color list](http://www.w3.org/TR/css3-color/#css2-system) as well.

Happy to update to [level 4](https://drafts.csswg.org/css-color-4/#named-colors) (though it's status is not yet recommended). The only change is the addition of `rebeccapurple` :+1:

### CODE REVIEW

File: grammars/sass.cson [hunk 1]
```diff
@@ -387,13 +387,13 @@
...
+        'comment': 'These colours are deprecated from CSS color module level 3 http://www.w3.org/TR/css3-color/#css2-system'
+        'match': '\\b(ActiveBorder|ActiveCaption|AppWorkspace|Background|ButtonFace|ButtonHighlight|ButtonShadow|ButtonText|CaptionText|GrayText|Highlight|HighlightText|InactiveBorder|InactiveCaption|InactiveCaptionText|InfoBackground|InfoText|Menu|MenuText|Scrollbar|ThreeDDarkShadow|ThreeDFace|ThreeDHighlight|ThreeDLightShadow|ThreeDShadow|Window|WindowFrame|WindowText)\\b'
```
Comment (50Wliu) [inline grammars/sass.cson]: These don't appear to be colors?

**Comment by bgriffith** [comment]
These are CSS system colors introduced in css 2 but deprecated by level 3 - http://www.w3.org/TR/css3-color/#css2-system

File: grammars/sass.cson [hunk 2]
```diff
@@ -387,13 +387,13 @@
...
         'name': 'invalid.deprecated.color.w3c-non-standard-color-name.css.sass'
```
Comment (50Wliu) [inline grammars/sass.cson]: Thanks for the clarification :+1:.  In that case, `w3c-non-standard-color-name` doesn't really seem to fit anymore - what about `invalid.deprecated.color.sytem.css.sass`?

**Comment by bgriffith** [comment]
Sounds good. I'll update the PR 👍

**Comment by bgriffith** [comment]
@50Wliu Will hopefully get a chance to look at this today. I looked at the `language-css` and noticed you have one rule for basic valid colors and one rule for extended valid colors. For continuity sake would you be happy for me to adopt the same in this PR?

File: grammars/sass.cson (hunk 2 above)
Comment (bgriffith) [inline grammars/sass.cson]: Just to confirm should that be:

`invalid.deprecated.color.system.scss`
`invalid.deprecated.color.system.sass`

or 

`invalid.deprecated.color.system.css.scss`
`invalid.deprecated.color.system.css.sass`

**Comment by 50Wliu** [comment]
There's really no consistency at all between the different CSS-type grammars, so that's your decision whether or not to do the same thing as language-css.

(Ideally everything would extend language-css but that's something for another day)

File: grammars/sass.cson (hunk 2 above)
Comment (50Wliu) [inline grammars/sass.cson]: Let's go with the second for now.

**Comment by bgriffith** [comment]
Ok updated deprecated rule name and left the valid color rulesets as they are.

**Comment by 50Wliu** [comment]
Thanks!

**Pull request merged**